bus-mapping/src/evm/opcodes/begin_end_tx.rs

```diff
@@ -615,3 +618,57 @@ fn gen_tx_l1_fee_ops(
     )?;
     Ok(())
 }
+
+// Add two copy-events for tx access-list addresses and storage keys if EIP-2930.
+fn gen_tx_eip2930_ops(
+    state: &mut CircuitInputStateRef,
+    exec_step: &mut ExecStep,
+) -> Result<(), Error> {
+    if !state.tx.tx_type.is_eip2930_tx() {
+        return Ok(());
+    }
+
+    let tx_id = NumberOrHash::Number(state.tx_ctx.id());
+    let (address_size, storage_key_size) =
+        access_list_address_and_storage_key_sizes(&state.tx.access_list);
+
+    // Add copy event for access-list addresses.
+    let rw_counter_start = state.block_ctx.rwc;
+    state.push_copy(
+        exec_step,
+        CopyEvent {
+            src_addr: 0,
+            src_addr_end: address_size,
+            src_type: CopyDataType::TxAccessListAddresses,
+            src_id: tx_id.clone(),
+            dst_addr: 0,
+            dst_type: CopyDataType::RwAccessListAddresses,
+            dst_id: tx_id.clone(),
+            log_id: None,
+            rw_counter_start,
+            // TODO
+            copy_bytes: CopyBytes::new(vec![], None, None),
```

Comment:
Is this the part that will be done in another PR?

Reply:
Yes, suppose to add in the following PR which adds the related code of copy-circuit.